src/rgw/rgw_file.h

```diff
@@ -682,8 +683,9 @@ namespace rgw {
     }
 
     void encode(buffer::list& bl) const {
-      ENCODE_START(2, 1, bl);
+      ENCODE_START(3, 1, bl);
       encode(uint32_t(fh.fh_type), bl);
+      encode(file_ondisk_version.load(), bl);
```

Comment:
New data in later versions of a structure need to be put at the end, the allowing an earlier RGW to decode this correctly.

Reply:
@ivancich Hi, I have to add it here, because when decoding, it relies on the decoded "file_ondisk_version" to estimate whether to go on decoding or just simple abandon the following fields.